### DISCUSSION TITLE
Required off-source UE4SS changes for upcoming UE5.5

### DISCUSSION THREAD

**Discussion by Kein** [UE4SS-RE/RE-UE4SS#672] [General]
Hey. It's been a while since I touched UE modding and UE4SS, I have just a few quick questions.
Outside of main offsets UE4SS requires, what other changes are needed for UE5.5 compatibility, assuming these changes can be made without source changes? My understanding is - I would also need an updated VTable to have ability to generate SDK dumps (this is all I'm after)? What does the said VTable describe? Looking at the example it seems to be some functions that are part of reflection data (and some hidden reflected stuff like `__vecDelDtor` ?).

**Comment by narknon** [comment]
The basic start is running UVTD on a source build PDB and implementing the generated files.  I think 5.5 did some changes to fstring and other things that may make it more difficult than usual to support.

**Reply by Kein** [reply]
This UVTD?
https://github.com/UE4SS-RE/RE-UVTD
```
[2024-10-05 01:16:01] Dumping 27 struct symbols for 5_05
[2024-10-05 01:16:01] Generating file 'VTableLayout_5_05_Template.ini'
[2024-10-05 01:16:01] Generating file 'GeneratedVirtualImplementations\generated_include\UnrealVirtual505.hpp'
[2024-10-05 01:16:01] Generating file 'GeneratedVirtualImplementations\generated_include\FunctionBodies\UnrealVirtual505.cpp'
```
I assume it needs to be updated for UE5.5 too? Seems like it fails to generate `VTableLayout_5_05_Template.ini` and the rest of the function  dont do anything at all.


Also, I looked over FString sources in 5.4 and 5.5 but there are 0 differences outside of some new includes for the Unity/non-Unity builds. What changes were you talking about?

**Reply by narknon** [reply]
No, the uvtd in this repo.  There is now a 5.05 branch with support.  You'll need to find a working ftext signature or make it optional.  The symbol for the function we typically use was not available in my pdb.

**Reply by Kein** [reply]
Oh, I wasn't talking about signatures for main function hooks, these I can find/update if needed. I was talking more in the context of any other supplementary changes needed for the dumps. I dont exactly know what VTable in UE4SS is used for (I assume pure its own modding functional), so I was asking if these are necessary to be updated for 5.5.
Right now, with all correct offsets, latest UE4SS exp. just outright crashes on `5.5-preview` and I can't really debug it sadly, because `xmake` build pipeline does not work on my machine and I tried to troubleshoot it **plenty**, before giving up.

> No, the uvtd in this repo. 

As mentioned above, I can't build yours , so instead, I've merged the changes from UE4SS repo to the original and built it, but still got an empty output. Tried both stripped and full PDBs.

**Comment by Kein** [comment]
It looks like latest experimental build no longer crashes, I havent looked at the changes yet. UE 5.5 version space seems to be not yet supported explicitly, and running in UE5.4 results in:

```
[2024-11-07 02:25:37.3488030] ProcessLocalScriptFunction is not available, the following features will be unavailable:
[2024-11-07 02:25:37.3492983] <Put function here responsible for creating custom UFunctions or events for BPs>
[...]
[2024-11-07 02:26:02.5683232] Could not generate property 'OptionalProperty /Script/CoreUObject.OptionalPropertyTestObject:OptionalString' because: Unsupported property class OptionalProperty
```

I assume these are still not supported for 5.5.x? Test was done on blank 5.5-preview1

**Reply by narknon** [reply]
There's literally a branch called 5_05support or something in UEPseudo

**Reply by Kein** [reply]
Oh lol, I saw it at a glance, but though it was old 5.0.5 EA not 5.5.x due to minor semver mismatch.

Are there any builds for that branch? Searching by branch reveals no pipeline actions in Actions for it.
